### DISCUSSION TITLE
More representative logo

### DISCUSSION THREAD

**Discussion by temeddix** [fzyzcjy/flutter_rust_bridge#1043] [Ideas]
Currently, the logo is only a telephone between a Rust crab and Flutter bird. Wouldn't it be nice to have an official logo like this?
![image](https://user-images.githubusercontent.com/66480156/217842356-81ba9b94-298b-4dce-b70b-161f8e8ca557.png)

Disclaimer: This was drawn by midjourney AI

**Comment by fzyzcjy** [comment]
That's interesting! But IMHO the original logo is also not ugly ;)

**Comment by 2bndy5** [comment]
I landed here because the current logo doesn't play well with dark theme since the background was removed https://github.com/fzyzcjy/flutter_rust_bridge/commit/c1ca05d0aa9526af8c82555435c35651a9a5ff27

Personally, I think the telephone image may be too antiquated in technological terms. I thought a Venn diagram would be more applicable:
<img width="450" alt="FRB" src="https://github.com/user-attachments/assets/adf99e51-c66f-49bc-9d9e-f1cda8aa63b2">

The green line that connects the intersection with the "Bridge" label is supposed to separate the "Flutter" and "Rust" labels, but the word "Flutter" is a bit long when using the same font (Roboto Mono) size as the rest of the labels.

This more compact design would better serve as a favicon in the docs as well:
<img width="128" alt="favicon" src="https://github.com/user-attachments/assets/db4f6521-4d39-4969-8975-ccf03033a485">

I'm not too confident about using the green color (`#049603`). Its just a distinguishable (hue) midpoint between the orange (`#f74c00`) and the blue (`#045b9f`). The intention was to pick colors that conformed with the flutter and rust branding while allowing a transparent background to work with either light or dark themes.

> [!tip]
> You can opt-in to [GitHub's Command palette](https://docs.github.com/en/get-started/using-github/github-command-palette) to quickly switch the site's theme on demand. This allows to see how the above logo displays in either light or dark themes.

Here's my WIP GIMP file I used to create the above images: [FRB_logo.zip](https://github.com/user-attachments/files/16924956/FRB_logo.zip)

---
And now for the legalese

- [Flutter's branding policy](https://docs.flutter.dev/brand)
- [Flutter's official logo assets](https://flutter.dev/brand)
- [Rust foundation's branding policy](https://foundation.rust-lang.org/policies/logo-policy-and-media-guide)
- I think the [rust-lang.org website's logo resources](https://github.com/rust-lang/www.rust-lang.org/tree/master/static/logos) would be the preferred/updated resource of rust logo assets.

**Reply by fzyzcjy** [reply]
That looks also interesting! However, I guess the original image may not be very ugly :)

**Reply by 2bndy5** [reply]
No one in this thread is calling the current logo ugly. That wouldn't be "constructive" (referring to a general code of conduct).

I'm simply suggesting an alternative that works with both light and dark theme.

Again, I think the telephone idea (in both OP and current logo) is an aging concept. And the fact that the current logo uses a black telephone with a black cable... well it looks like this in dark theme:
![image](https://github.com/user-attachments/assets/879bd719-de0d-44b7-a8f0-b8b48424baee)

**Reply by patmuk** [reply]
> No one in this thread is calling the current logo ugly. That wouldn't be "constructive" (referring to a general code of conduct).
> 
> I'm simply suggesting an alternative that works with both light and dark theme.
> 
> Again, I think the telephone idea (in both OP and current logo) is an aging concept. And the fact that the current logo uses a black telephone with a black cable... well it looks like this in dark theme: ![image](https://private-user-images.githubusercontent.com/14963867/365538860-879bd719-de0d-44b7-a8f0-b8b48424baee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4NzgxNTcsIm5iZiI6MTcyNTg3Nzg1NywicGF0aCI6Ii8xNDk2Mzg2Ny8zNjU1Mzg4NjAtODc5YmQ3MTktZGUwZC00NGI3LWE4ZjAtYjhiNDg0MjRiYWVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA5VDEwMzA1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY1YzJkMWZjOWQ5MjdkMzE2ZTJkYzVkNzVjNGYzMjM4MmI1NWFmMDNmZDk5OWI1YzZjNzVmNjNmNmY0NWE5MWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LI1FOQMbWhVYD30_LXC8cJgYAygdm5H2fhXIVwRR1Hg)

why not tweaking the original logo? Having another color for the telephone - maybe that matrix-ish green.

**Reply by 2bndy5** [reply]
> why not tweaking the original logo?

Because I wanted something that could also be down-scaled to a favicon. Currently, the docs just use a blue dot:
![image](https://github.com/user-attachments/assets/fdc13ad7-57dc-40af-9955-72f399cd8aae)

> Having another color for the telephone

As I mentioned twice before, the telephone (picture of a land-line handset) is an aging technology. IMHO, it does not suite this project which uses technology that only recently has turned 10 years old.

**Comment by 2bndy5** [comment]
> Disclaimer: This was drawn by midjourney AI

@temeddix We should be careful when using AI to generate images for branding. Currently, most of the AI tools available to us do/may not respect copyright or trademark laws. While this isn't the end-users' problem, it could mean an unforeseen legal action taken against end-users in the future.
